test/distributed/test_c10d.py

```diff
@@ -4639,6 +4640,48 @@ def test_broadcast_coalesced_gloo_cpu(self):
         for root_rank in ranks:
             self._test_broadcast_coalesced(process_group, device, root_rank)
 
+    @requires_gloo()
+    def test_pass_gloo_options(self):
+        pg_opts = c10d.ProcessGroupGloo.Options()
+        pg_opts.devices = [create_device(interface=LOOPBACK)]
+        pg_opts.timeout = timedelta(seconds=1)
+        pg_opts.threads = 2
+
+        dist.init_process_group(
+            "gloo",
+            init_method=f"file://{self.file_name}",
+            world_size=self.world_size,
+            rank=self.rank,
+            pg_options=pg_opts
+        )
+
+        default_pg = c10d.distributed_c10d._get_default_group()
+        dist.destroy_process_group(default_pg)
+        self.assertFalse(dist.is_initialized())
+
+    @requires_nccl()
+    @skip_if_not_multigpu
+    def test_pass_nccl_options_high_priority_stream(self):
+        pg_opts = ProcessGroupNCCL.Options()
+        pg_opts.is_high_priority_stream = True
+
+        dist.init_process_group(
+            "nccl",
+            init_method=f"file://{self.file_name}",
+            world_size=self.world_size,
+            rank=self.rank,
+            pg_options=pg
+        )
+
+        default_pg = c10d.distributed_c10d._get_default_group()
+
+        # Test with new_group
+        pg = c10d.new_group([0, 1], pg_options=pg_opts)
+        t = torch.tensor([self.rank + 1] * 10).cuda(2 * self.rank)
+        pg.allreduce(t).wait()
+        expected_tensor = torch.tensor([3] * 10).cuda(2 * self.rank)
+        self.assertEqual(expected_tensor, t)
```

Comment:
Does this just verify allreduce is correct or does it also somehow verify we're using a high-pri stream?